main.rb

```diff
@@ -0,0 +1,244 @@
+
+  if permission_status == 'Y'
+    new_student = Student.new(student_class, student_age, student_name)
+  else
+    parent_status = false
+    new_student = Student.new(student_class, student_age, student_name, parent_status)
+  end
+  sleep(1)
+  print "Person created successfully\n"
+  $person_list << new_student
+  puts
+  main
+end
+
+def create_new_teacher
+  print 'Age: '
+  teacher_age = gets.chomp.to_i
+  print 'Name: '
+  teacher_name = gets.chomp
+  print 'Specialization: '
+  teacher_speacialization = gets.chomp
+  new_teacher = Teacher.new(teacher_speacialization, teacher_age, teacher_name)
+  sleep(1)
+  print "Person created successfully\n"
+  $person_list << new_teacher
+  puts
+  main
+end
+
+def create_person_obj(student_or_person)
+  if student_or_person == '1'
+    create_new_student
+  else
+    create_new_teacher
+  end
+end
+
+# Methods for creating student or teacher object ends here
+
+# Create book methods
+def create_book
+  print 'Title: '
+  book_title = gets.chomp
+  print 'Author: '
+  book_author = gets.chomp
+  new_book = Books.new(book_title, book_author)
+  sleep(1)
+  print "Book created successfully\n"
+  $book_list << new_book
+  puts
+  main
+end
+
+# Create book method end
+
+# Create rental
+
+def book_options
+  $book_list.each_with_index do |item, index|
+    puts "#{index}) Title: #{item.book}, Author: #{item.author}"
+    puts
+  end
+  book_number = gets.chomp.to_i
+  if book_number > $book_list.length
+    puts 'Please choose from the list'
+    book_options
+  end
+  book_number
+end
+
+def person_options
+  $person_list.each_with_index do |item, index|
+    puts "#{index}) [#{item.type}] Name: #{item.name}, id: #{item.id} Age: #{item.age}"
+    puts
+  end
+  person_number = gets.chomp.to_i
+  if person_number > $person_list.length
+    puts 'Please choose from the list'
+    person_options
+  end
+  person_number
+end
+
+def create_rental
+  if $book_list[0] and $person_list[0]
+    rent_book = book_options
+    rent_to = person_options
+    puts 'Insert date in the format YYYY-DD-MM'
+    print 'Date:'
+    enter_date = gets.chomp
+    new_rental = Rental.new(enter_date, $book_list[rent_book], $person_list[rent_to])
+    puts 'Rental Created Successfully'
+    $rentals << new_rental
+    puts
+    main
+  end
+  puts 'Not enough books or students found'
+  puts
+  main
+end
+
+def show_books
+  puts 'No books added yet. Press (4) to add a book.' if $book_list.none?
+  puts
+  if $book_list.length.positive?
+    puts 'Book List'
+    $book_list.each do |item|
+      puts "Book: #{item.book}, Author: #{item.author}"
+      puts
+    end
+  end
+  puts
+  main
+end
+
+def show_person
+  puts 'Nothing added to person list yet. Press (3) to add a person' if $person_list.none?
+  puts
+  if $person_list.length.positive?
+    puts 'Person List'
+    $person_list.each do |val|
+      puts
+      print "#{val.type}-ID: (#{val.id}): Name: #{val.name} Age: #{val.age} Allowed: #{val.permission}"
```

Comment:
- [ ] A person's _name_ and _age_ are **not being listed correctly**. The _name_ is interchanging with the _age_ in your app. Kindly ensure that the _name_ and _age_ lists the correct data.

**Persons list error:**
![image](https://user-images.githubusercontent.com/12745474/141611292-aba38b71-78e2-4c88-82b6-71359f6869b6.png)